lib/valueflow.cpp

```diff
@@ -2920,6 +2920,18 @@ static bool valueFlowForwardVariable(Token* const startToken,
                 return false;
         }
 
+        // Function call
+        else if (!var->isLocal() && tok2->function() && Token::Match(tok2, "%name% (")) {
+            if (tok2->function()->functionScope) {
+                const bool c = isVariableChanged(
+                                   tok2->function()->functionScope->bodyStart,
+                                   tok2->function()->functionScope->bodyEnd, varid,
+                                   var->isGlobal(), settings, tokenlist->isCPP());
+                if (c)
+                    lowerToInconclusive(values, settings);
```

Comment:
It should not be inconclusive if we know the variable has changed. It should be inconclusive if the variable hasn't changed in the function and the function has called a non-const member function; or its inconclusive when the `functionScope` is not available and `!tok->function()->isConst()`.

Reply:
Done, thanks for the review !